frontend/src/concepts/modelRegistry/utils.ts

```diff
@@ -113,3 +114,80 @@ export const filterArchiveModels = (registeredModels: RegisteredModel[]): Regist
 
 export const filterLiveModels = (registeredModels: RegisteredModel[]): RegisteredModel[] =>
   registeredModels.filter((rm) => rm.state === ModelState.LIVE);
+
+/**
+ * Converts model source properties to catalog parameters
+ * @param modelSourceKind - The kind of model source
+ * @param modelSourceClass - The class of model source (maps to catalog source name)
+ * @param modelSourceGroup - The group of model source (maps to catalog repository name)
+ * @param modelSourceName - The name of model source (maps to catalog model name)
+ * @param modelSourceId - The ID of model source (maps to catalog model tag)
+ * @returns CatalogModelDetailsParams object or null if not a catalog source
+ */
+export const modelSourcePropertiesToCatalogParams = (
+  modelSourceKind?: string,
+  modelSourceClass?: string,
+  modelSourceGroup?: string,
+  modelSourceName?: string,
+  modelSourceId?: string,
```

Comment:
To make this easier, maybe we should factor out a type from ModelArtifact for `ModelSourceProperties` that contains these, and then ModelArtifact can inherit them with `&` like it does with ModelRegistryBase. Then you can just have this function take a `ModelSourceProperties` object, and passing the whole ModelArtifact into this function will be valid since it's a superset of that type.